corgi/core/models.py

```diff
@@ -1046,7 +1046,7 @@ def manifest_components(self, ofuri: str, quick=False) -> "ComponentQuerySet":
         non_container_source_components = self.exclude(name__endswith="-container-source").using(
             "read_only"
         )
-        roots = non_container_source_components.root_components()
+        roots = non_container_source_components.root_components().exclude(MODULAR_SRPM_CONDITION)
```

Comment:
This has the effect of hiding both the non-modular and modular SRPM from the manifest. I think it's because of the interaction between root_components and latest_components. My understanding is that the root_components condition here is calculating which packages to gather for the latest_component stored procedure, so if you exclude modular SRPMs here, it's still present in the latest_components calculation. In this example of apache-common-io, I think the root_component condition is correctly finding 'apache-commons-io:1-2.8.0-7.el9.src' (and excluding the modular SRPM) but when the latest_components function runs it again calls root_components (on line 943 of models.py) and includes the modular SRPM, so it get it's component UUID as the latest, but doesn't find that in the original query so filters both out.

Reply:
@jasinner re comment about root_components and latest components ... sounds reasonable to me